### PR TITLE
RF and LogReg: add class_weight parameter

### PR DESCRIPTION
**Pull request by ajdapretnar**
##### Issue

Implements #4986.

##### Description of changes
Add checkbox to consider class weights in Logistic Regression and Random Forest.
Trigger Warning when checking the parameter.
Add tests.

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by ajdapretnar** [comment]
I don't like tests. They seem useless. Any suggestion?

**Comment by janezd** [comment]
Tests test scikit-learn, not Orange: they test whether the flag has any effect (which happens in sklearn) and not whether the widget actually sets the flag.

Add a test to test_owlogisticregression and test_owrandomforest, where you set and unset the flag and check whether the learner on the output (don't give any data to the widget) has the `weight_class` argument properly set. You can also check whether the warning properly appears and disappears.

The tests that you have added should better be removed. Not only because they don't test our code, but also because they are a bit dangerous.

Second, the label at the checkbox, "Class weight" is not very informative, and the tooltip is a bit misleading -- classes are weighed according to the *inverse* proportion of frequencies. Also, for logistic regression, I'd prefer a separate box because the existing box is focused on regularization parameters.

Screenshot 2020-10-16 at 11 15 35

I experimented with UX changes, so I already have the code and I'll just push a commit. I'd ask you just to fix the tests.

**Comment by janezd** [comment]
I'm so sorry, I accidentally rebased before pushing my commit. Please ask your local git expert how to pull your branch before adding or changing your commits. Squash at will.

**Comment by ajdapretnar** [comment]
I did feel weird about those tests, but just checking a flag seemed... as if I haven't done anything. 😆 But you are right, I shouldn't be checking sklearn's code.

I also welcome the UX changes. My local git expert did his job well. ;)

**Comment by janezd** [comment]
> just checking a flag seemed... as if I haven't done anything.

That's because your code hasn't done anything (almost). The code's job is to set the flag, and the test's job is to verify that the code does it's job. Reacting to the flag is up to sklearn, and testing the reaction is up to sklearn's tests.

File: Orange/widgets/model/tests/test_owlogisticregression.py [hunk 1]
```diff
@@ -112,3 +112,13 @@ def test_target_with_nan(self):
...
+
+    def test_class_weights(self):
+        table = Table("iris")
+        self.send_signal("Data", table)
+        self.assertFalse(self.widget.class_weight)
+        self.widget.controls.class_weight.setChecked(True)
+        self.assertTrue(self.widget.class_weight)
+        self.widget.apply_button.button.click()
+        self.assertEqual(self.widget.model.skl_model.class_weight, "balanced")
+        self.assertTrue(self.widget.Warning.class_weights_used.is_shown())
```
Comment (janezd) [inline Orange/widgets/model/tests/test_owlogisticregression.py:124]: I won't nitpick further, but the more important test, IMHO, is that the warning is cleared when necessary. We are good at raising them, but used to have problems clearing.

You also haven't tested that the argument is not set when the checkbox is unchecked. But let's assume that this but is unlikely, so I'm merging. :)
